File: Data Structures & Algorithms/duplicate-integer/submission-9.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let _set = new Set()
        for (const num of nums){
            if (_set.has(num)){
                return true 
            }
            _set.add(num);
        }
        return false 
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        frequencyArray = [[] for i in range(len(nums) + 1)]
        hashmap = {}
        for num in nums: 
            hashmap[num] = hashmap.get(num, 0) + 1
        print(hashmap)
        print(frequencyArray)
        for keys in hashmap:
            frequencyArray[hashmap[keys]].append(keys)
        
        res = []
        for i in range(len(frequencyArray) - 1, -1, -1):
            for elements in frequencyArray[i]:
                res.append(elements)
                if len(res) == k:
                    return res 
        return res 
